client/app/core/github.service.ts

```diff
@@ -1,20 +1,20 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable } from 'rxjs/Observable';
+import { ApiIssue } from './github-issue.model';
 
 @Injectable()
 export class GithubService {
 
   constructor(private http: HttpClient) { }
 
-  getIssue(url: string): Observable<any> {
+  /** Return an issue or pr from the GitHub API */
+  getIssue(url: string): Observable<ApiIssue> {
     return this.http.get(url)
       .map((result: any) => {
-        let type = 'issue';
-        if (result.pull_request || result.merged !== undefined) {
-          type = 'pr';
-        }
-        return { type: type, data: result };
+        const type = result.pull_request || result.merged !== undefined ? 'pr' : 'issue';
```

Comment:
Should probably do one of the following

1. If `pull_request`, then change the value of the input and get the actual pull request
2. Just treat issue-fetched PRs as issues